Feature/add support for resource types wms layer and vmts layer

sweco-se1g7y opened the pull request:
WMS and WMTS layers should now be correctly processed.

Resolve #67

sweco-se1g7y committed:
build: updated dependencies

sweco-se1g7y committed:
feat: added support for WMT and WMTS layers

sweco-se1g7y committed:
feat: added error handling

sweco-se1g7y committed:
fix: removed namespaces from Geoserver

sweco-se1g7y committed:
feat: generalised handling of geoserver entities

sweco-sedalh commented on plugins/catalog-backend-module-geoserver/src/GeoserverDataProvider.ts:
```diff
@@ -270,27 +261,30 @@ export class GeoserverDataProvider implements EntityProvider {
+                },
+              },
+              spec: {
+                dialect: 'geoserver',
```
Dialect isn't really needed here since we already know it's from a geoserver (since the kind is GeoserverLayer).

It might make sense to make this more general ("MapServiceLayer"?), but I'm not quite sure how to properly model that right now, so I think going with GeoserverLayer (and without dialect) is fine for now!

sweco-se1g7y committed:
fix: made sure all entered Geoserver entity name follow Backstage requirements

fix: ensured compliant names are used everywhere in the Geoserver data provider

sweco-se1g7y committed:
feat: removed dialect from Geoserver entities

sweco-se1g7y committed:
fix: fixed bug which caused some names to be rejected

sweco-se1g7y committed:
feature: changed Geoserver titles to equal original names
